### ISSUE TITLE
[Bug]: The Tab pickup CFR cannot be dismissed while the search bar is active

### ISSUE THREAD

**Issue by AdinaPetridean** [mozilla-mobile/fenix#26957]
### Prerequisites: 
On Desktop or another mobile device visit some webpages making sure you are signed in with the same account.

### Steps to reproduce
1. Perform a clear data and launch the Firefox application.
2. From the "Pick up where you left" section tap on Sign in button and sign in with the same account.
3. Dismiss the Onboarding.
4. While the search bar is active, scroll up.
5. Once the Synced tabs section is displayed, tap outside the "Your tabs are syncing! Pick up where you left off on your other device" message or tap the "X" button from the CFR. 
6. Observe the behavior.

### Expected behaviour

The Tab pickup CFR is dismissed when tapping outside the prompt or when tapping the "X" button.

### Actual behaviour

The Tab pickup CFR cannot be dismissed by tapping outside the prompt or when tapping the "X" button while the search bar is active.

### Device name

OnePlus 9 Pro  

### Android version

Android 12

### Firefox release type

Firefox Nightly

### Firefox version

Firefox Nightly 106.0a1 - 09/13

### Additional information

Please note this issue is also reproducible with Poco M4 Pro (Android 12), Xiaomi Pad5 (Android 11) and other devices.

https://user-images.githubusercontent.com/89388888/190157865-de640b18-2efa-47ec-a8b6-04b04e6e70bd.mp4

https://user-images.githubusercontent.com/108326843/189911366-58f7fe10-502a-410e-9d67-876be85113dc.mp4

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21553)

**Comment by delia-pop** [comment]
Upon further testing we noticed that this issue is caused by the active search bar, as it is not reproducible when the search dialog is not active. We updated the STR accordingly.
We discussed this with @Mugurell and he explained that the fix for https://github.com/mozilla-mobile/fenix/issues/26690 might not solve the CFR dismissal issue as it is.

**Comment by Mugurell** [comment]
The fix for https://github.com/mozilla-mobile/fenix/issues/26690 solves that specific scenario while this seems like another one related.
To fix this bug and maybe others I think we have to address the common cause - the toolbar remaining in edit mode and so it having focus and not the homescreen content.
So an easy fix would be to exit edit mode when the keyboard is hidden.

**Comment by cpeterso** [comment]
When the CFR can't be dismissed, can the user still use the search bar and the rest of the app? Or is the whole app stuck?

If the user can still use the search bar, then this bug is a lower priority and doesn't need to be fixed in MR v106.

If the whole app is stuck, then this bug is a high priority for MR v106.

**Comment by Mugurell** [comment]
This bug affects all popups showing all screen - other CFRs or menus appearing after the toolbar is in edit mode.
The issue is encountered with the "Tab pickup CFR" because there's an inherent delay to download and shows the synced tab and it is possible for it (and the CFR) to be shown after the user already interacted with the toolbar and left it in edit mode.
Users can still interact with the app - enter an URL to visit or tap the other buttons from the toolbar - first touch will dismiss the CFR.


https://user-images.githubusercontent.com/11428869/193053649-ff47fdce-2c0f-419f-8623-ea23359d1c79.mp4

**Comment by Mugurell** [comment]
The bigger issue with the toolbar intercepting touch events was half-way avoided - just for the menus in https://github.com/mozilla-mobile/fenix/pull/26832.
To fix it for CFRs also (and simplify the code a bit by not having to account for different touch scenarios) we could make a decision to not leave the toolbar in edit mode when the homescreen is shown.
The above PR shows how this works when long pressing on items that shows the "Remove" menu. We'd do that by default.

**Comment by Mugurell** [comment]
From @cpeterso 
> Sounds like we don't need to rush a fix for 106 but we should clean up this code (like your suggestion to exit edit mode when the keyboard is closed).

**Comment by alicevarro** [comment]
Using [Firefox RC 106.1.0](https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/mobile.v2.fenix.release.2022.10.11.latest.arm64-v8a/artifacts/public%2Fbuild%2Farm64-v8a%2Ftarget.apk) (Build ID #2015909131) on One Plus 6 (Android 11) I was able to reproduce this issue and found another behavior that seems related to this using the following STR:
1. Launch Fenix.
2. Tap the "Sign in" button from the "Pick up where you left off" card.
3. Sign in with an existing account (I tried both the QR code scan and “Use email instead” option).
4. Dismiss the Onboarding Tour.
5. Scroll up to see the "Jump back in" section
6. Scroll down to the bottom of the page.

The Tab Pickup CFR is moving towards the bottom of the page and it is dismissed around the “Thought-provoking stories area”.
Attaching a screen recording of the behavior:

https://user-images.githubusercontent.com/96465757/195592212-c2314e3b-d4c6-4241-9330-32f1374f3008.mov

**Comment by AdinaPetridean** [comment]
While retesting this issue on OnePlus 9 Pro (Android 12) the following behaviours were observed:
1. When the user taps outside, the Tab Pickup CFR is dismissed only after the second attempt. Please observe the attached video. 

https://user-images.githubusercontent.com/108326843/207558913-1f8ff460-c00b-4d0a-966f-8590875e4050.mp4

2. When the user taps the "X" button, the "Pair using an app" page is displayed.  Observe the attached video for more information:

https://user-images.githubusercontent.com/108326843/207558958-8c79c835-4ce7-4c71-a84c-91c50c41c996.mp4

@Alexandru2909

**Comment by Alexandru2909** [comment]
Hi @AdinaPetridean, thanks for testing this! I'm working on a fix for the issue described in your comment.

**Comment by delia-pop** [comment]
Verified as fixed on the latest Nightly 110.0a1 from 12/24 with OPPO A15s (Android 10). The Jump back in and Tab Pickup CFRs can now be dismissed. 
The behavior described in the [comment](https://github.com/mozilla-mobile/fenix/issues/26957#issuecomment-1350729485) above is intended: the first tap (or attempt to swipe on screen will make the search bar inactive, and the 2nd tap on screen will  dismiss the CFR).
